service/controller/app/v1/resource/chart/create_test.go

```diff
@@ -0,0 +1,198 @@
+package chart
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned/fake"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/micrologger/microloggertest"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/kubeconfig"
+)
+
+func TestResource_newCreateChange(t *testing.T) {
+	type fields struct {
+		g8sClient      versioned.Interface
+		k8sClient      kubernetes.Interface
+		kubeConfig     *kubeconfig.KubeConfig
+		logger         micrologger.Logger
+		watchNamespace string
+	}
+	tests := []struct {
+		name            string
+		currentResource *v1alpha1.Chart
+		desiredResource *v1alpha1.Chart
+		expectedChart   *v1alpha1.Chart
+	}{
+		{
+			name:            "case 0: new chart should be created",
+			currentResource: &v1alpha1.Chart{},
+			desiredResource: &v1alpha1.Chart{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Chart",
+					APIVersion: "application.giantswarm.io",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "kubernetes-prometheus",
```

Comment:
```suggestion
					Name: "prometheus",
```

We're dropping this prefix as its not really useful.